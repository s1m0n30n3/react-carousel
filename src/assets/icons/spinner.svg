<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 100 100">
<rect x="19" y="26" width="12" height="48" fill="#787878">
  <animate attributeName="y" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="2;26;26" keySplines="0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.22222222222222224s"></animate>
  <animate attributeName="height" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="96;48;48" keySplines="0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.22222222222222224s"></animate>
</rect>
<rect x="44" y="26" width="12" height="48" fill="#787878">
  <animate attributeName="y" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="8;26;26" keySplines="0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.11111111111111112s"></animate>
  <animate attributeName="height" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="84;48;48" keySplines="0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.11111111111111112s"></animate>
</rect>
<rect x="69" y="26" width="12" height="48" fill="#787878">
  <animate attributeName="y" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="8;26;26" keySplines="0 0.5 0.5 1;0 0.5 0.5 1"></animate>
  <animate attributeName="height" repeatCount="indefinite" dur="1.1111111111111112s" calcMode="spline" keyTimes="0;0.5;1" values="84;48;48" keySplines="0 0.5 0.5 1;0 0.5 0.5 1"></animate>
</rect>
</svg>