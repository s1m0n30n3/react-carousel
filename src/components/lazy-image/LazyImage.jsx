import React, { useLayoutEffect } from "react";

import { useLazyLoading } from "hooks";
import { lazyImagepropTypes } from "proptypes/index";

export const LazyImage = ({ src, imageLoad, ...props }) => {
  const { icon: Icon, size } = imageLoad;
  const { onImageLoad, isLoaded } = useLazyLoading({ src });

  useLayoutEffect(() => {
    const initialImage = new Image();
    initialImage.src = src;
    onImageLoad(initialImage);
  }, [src, onImageLoad]);

  if (!isLoaded) return <Icon fontSize={size} />;
  return <img src={src} {...props} />;
};

LazyImage.propTypes = lazyImagepropTypes;
LazyImage.displayName = "Gallery.LazyImage";
