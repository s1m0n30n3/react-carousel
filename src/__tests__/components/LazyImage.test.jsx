import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, test } from "vitest";

import { LazyImage } from "components/lazy-image";
import { ReactComponent as LoadingSpinner } from "assets/icons/spinner.svg";

describe("LazyImage", () => {
  test("renders correctly", () => {
    const component = renderer.create(
      <LazyImage
        imageLoad={{
          icon: LoadingSpinner,
          size: 24,
        }}
        src="image.png"
        alt="alt text"
        draggable={false}
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
