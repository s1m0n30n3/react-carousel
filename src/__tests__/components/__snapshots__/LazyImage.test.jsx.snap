// Vitest Snapshot v1

exports[`LazyImage > renders correctly 1`] = `
<svg
  fontSize={24}
  height="1em"
  viewBox="0 0 100 100"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#787878"
    height={48}
    width={12}
    x={19}
    y={26}
  >
    <animate
      attributeName="y"
      begin="-0.22222222222222224s"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="2;26;26"
    />
    <animate
      attributeName="height"
      begin="-0.22222222222222224s"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="96;48;48"
    />
  </rect>
  <rect
    fill="#787878"
    height={48}
    width={12}
    x={44}
    y={26}
  >
    <animate
      attributeName="y"
      begin="-0.11111111111111112s"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="8;26;26"
    />
    <animate
      attributeName="height"
      begin="-0.11111111111111112s"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="84;48;48"
    />
  </rect>
  <rect
    fill="#787878"
    height={48}
    width={12}
    x={69}
    y={26}
  >
    <animate
      attributeName="y"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="8;26;26"
    />
    <animate
      attributeName="height"
      calcMode="spline"
      dur="1.1111111111111112s"
      keySplines="0 0.5 0.5 1;0 0.5 0.5 1"
      keyTimes="0;0.5;1"
      repeatCount="indefinite"
      values="84;48;48"
    />
  </rect>
</svg>
`;
